tests/test_count_character_frequency.py

```diff
@@ -0,0 +1,77 @@
+import os
+from collections import Counter
+from pathlib import Path
+from typing import Generator
+
+import pytest
+from _pytest.tmpdir import TempPathFactory
+
+from characters_counter import count_character_frequency
+
+CONTENT = (
+    "Typer is a library for building CLI applications that users will love using and developers "
+    "will love creating. Typer is a library for building CLI applications that users will love using and "
+    "developers will love creating. Based on Python type hints. "
+    "It's also a command line tool to run scripts, automatically converting them to CLI applications."
+)
+
+
+class TestCharacterFrequency:
+    @pytest.mark.parametrize(
+        "data_input, result",
+        [
+            ("Python is the best language in the world", {"t": 4, "e": 4, "h": 3}),
+            ("python", {"p": 1, "y": 1, "t": 1}),
+            ("", {}),
+        ],
+    )
+    def test_count_character_frequency_with_string_input(self, data_input: Path, result: dict) -> None:
+        assert count_character_frequency(data_input) == result
+
+    @pytest.fixture(scope="function")
+    def sample_file(self, tmp_path_factory: TempPathFactory) -> Generator[Path, None, None]:
+        tmp_file_path = tmp_path_factory.mktemp("data") / "text_sample.txt"
+        tmp_file_path.write_text(CONTENT)
+        yield tmp_file_path
+        os.remove(tmp_file_path)
+
+    def test_count_character_frequency_with_correct_filepath_input_using_pytest_fixture(
+        self, sample_file: Path
+    ) -> None:
+        result = count_character_frequency(sample_file)
+        expected_result = {
```

Comment:
That seems to be repetitive, maybe you can utilize class variable somehow

Reply:
It makes a sense, okey